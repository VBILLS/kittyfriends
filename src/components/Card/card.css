.card {
  box-sizing: border-box;
  display: inline-block;
  background-color: rgba(255, 158, 236, 0.41);
  border-radius: 0.5rem;
  padding: 1rem;
  margin: 0.5rem;
  box-shadow: 4px 4px 8px 0px rgba(0, 0, 0, 0.2);
}

.grow {
  -moz-osx-font-smoothing: grayscale;
  backface-visibility: hidden;
  transform: translateZ(0);
  transition: transform 0.25s ease-out;
}

.grow:hover,
.grow:focus {
  transform: scale(1.05);
}
/*  @media screen and (max-width: 400px) {
    width: 50vw;
  } */

.text {
  font-size: 12px;
  text-overflow: clip;
  word-wrap: inherit;
}
