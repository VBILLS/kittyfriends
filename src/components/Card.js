import React from 'react';

import './card.css';

const Card = ({ name, email, id }) => {
  return (
    <div className='card tc grow br3 pa3 ma2 dib bw2 shadow-5'>
      <img
        alt='robots'
        src={`https://robohash.org/${id}?size=200x200&set=set4`}
      />
      <div>
        <h2>{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  );
};

export default Card;
