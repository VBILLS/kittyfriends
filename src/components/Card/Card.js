import React from 'react';

import './card.css';

const Card = ({ name, email, id }) => {
  return (
    <div className='card grow'>
      <img
        alt='robots'
        src={`https://robohash.org/${id}?size=200x200&set=set4`}
      />
      <div>
        <h2>{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  );
};

export default Card;
