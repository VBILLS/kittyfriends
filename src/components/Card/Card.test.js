import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import Card from './Card';

describe('Card component', () => {
  it('matches react-test-renderer snapshot', () => {
    const component = renderer.create(<Card />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('shallow renders & matches snapshot with enzyme', () => {
    const wrapper = shallow(<Card />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders props', () => {
    const mockProps = {
      name: 'VCunit',
      email: 'vonacunt@gmail.com'
    };
    const newWrapper = shallow(<Card {...mockProps} />);
    expect(toJson(newWrapper)).toMatchSnapshot();
  });
});
