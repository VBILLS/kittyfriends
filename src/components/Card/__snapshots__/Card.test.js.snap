// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component matches react-test-renderer snapshot 1`] = `
<div
  className="card grow"
>
  <img
    alt="robots"
    src="https://robohash.org/undefined?size=200x200&set=set4"
  />
  <div
    className="text"
  >
    <h2 />
    <p />
  </div>
</div>
`;

exports[`Card component renders props 1`] = `
<div
  className="card grow"
>
  <img
    alt="robots"
    src="https://robohash.org/undefined?size=200x200&set=set4"
  />
  <div
    className="text"
  >
    <h2>
      VCunit
    </h2>
    <p>
      vonacunt@gmail.com
    </p>
  </div>
</div>
`;

exports[`Card component shallow renders & matches snapshot with enzyme 1`] = `
<div
  className="card grow"
>
  <img
    alt="robots"
    src="https://robohash.org/undefined?size=200x200&set=set4"
  />
  <div
    className="text"
  >
    <h2 />
    <p />
  </div>
</div>
`;
